integration-libs/epd-visualization/components/visual-viewer/visual-viewer.service.ts

```diff
@@ -0,0 +1,1498 @@
+import {
+  ChangeDetectorRef,
+  ElementRef,
+  EventEmitter,
+  Injectable,
+  OnDestroy,
+} from '@angular/core';
+// @ts-ignore
```

Comment:
Can we add a code comment, why this @ts-ignore is needed here?

Reply:
I have added a comment.
Without the `// @ts-ignore` the following compile error is produced:

```text
An unhandled exception occurred: integration-libs/epd-visualization/components/visual-viewer/visual-viewer.service.ts:8:22 - error TS2306: File '/Users/i070295/SAPDevelop/sap/spartacus/node_modules/@sapui5/ts-types-esm/types/index.d.ts' is not a module.

8 import * as ui5 from '@sapui5/ts-types-esm';
                       ~~~~~~~~~~~~~~~~~~~~~~
```

If the` import * as ui5 from '@sapui5/ts-types-esm';` is removed none of the UI5 tyypes can be resolved:
`An unhandled exception occurred: integration-libs/epd-visualization/components/visual-viewer/visual-viewer.service.ts:31:23 - error TS2307: Cannot find module 'sap/ui/core/Core' or its corresponding type declarations.

31 import type Core from 'sap/ui/core/Core';
`

Reply:
Understood, organizeImports can be annoying in this case. But IMHO avoiding this inconvenience for Spartacus developers is not enough reason to force all customers using the feature to degrade type-safety by setting `skipLibCheck` for them.

As an alternative to a fake import `import * as ui5 from '@sapui5/ts-types-esm'` (which is semantically incorrect, as ui5 is not a module), did you consider declaring a [dependency on a package](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-) in the top of file? It could be preceded with an explanation comment:
```
// '@sapui5/ts-types-esm' package contains types for sap modules, e.g. 'sap/ui/core/Core'
/// <reference types="@sapui5/ts-types-esm" />
```